services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -282,9 +290,15 @@ public Supplier<Map<String, Object>> heartbeatDimensions(Task task)
 
           @Provides
           @LazySingleton
-          public Task readTask(@Json ObjectMapper mapper, ExecutorLifecycleConfig config)
+          public Task readTask(@Json ObjectMapper mapper, @Smile ObjectMapper smileMapper, ExecutorLifecycleConfig config, TaskPayloadManager taskPayloadManager, TaskConfig taskConfig)
```

Comment:
What's the reason to use a Smile mapper? Seems same as Json mapper.

Reply:
we use the smile mapper to serialize the payload so i figured we should use the same one to deserialize